webhooks/cluster.go

```diff
@@ -52,54 +61,210 @@ func (c *Cluster) Default(_ context.Context, obj runtime.Object) error {
+				field.NewPath("spec", "topology"),
+				"can be set only if the ClusterTopology feature flag is enabled",
+			),
+		}
+	}
+
+	var allErrs field.ErrorList
+
+	// class should be defined.
+	if len(new.Spec.Topology.Class) == 0 {
+		allErrs = append(
+			allErrs,
+			field.Invalid(
+				field.NewPath("spec", "topology", "class"),
+				new.Spec.Topology.Class,
+				"cannot be empty",
+			),
+		)
+	}
+
+	// version should be valid.
+	if !version.KubeSemver.MatchString(new.Spec.Topology.Version) {
+		allErrs = append(
+			allErrs,
+			field.Invalid(
+				field.NewPath("spec", "topology", "version"),
+				new.Spec.Topology.Version,
+				"must be a valid semantic version",
+			),
+		)
+	}
+
+	// MachineDeployment names must be unique.
+	if new.Spec.Topology.Workers != nil {
+		names := sets.String{}
+		for _, md := range new.Spec.Topology.Workers.MachineDeployments {
+			if names.Has(md.Name) {
+				allErrs = append(allErrs,
+					field.Invalid(
+						field.NewPath("spec", "topology", "workers", "machineDeployments"),
+						md,
+						fmt.Sprintf("MachineDeployment names should be unique. MachineDeployment with name %q is defined more than once.", md.Name),
+					),
+				)
+			}
+			names.Insert(md.Name)
+		}
+	}
+
+	if old != nil { // On update
+		// Class could not be mutated.
+		if new.Spec.Topology.Class != old.Spec.Topology.Class {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "topology", "class"),
+					new.Spec.Topology.Class,
+					"class cannot be changed",
+				),
+			)
+		}
+
+		// Version could only be increased.
+		inVersion, err := semver.ParseTolerant(new.Spec.Topology.Version)
+		if err != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "topology", "version"),
+					new.Spec.Topology.Version,
+					"is not a valid version",
+				),
+			)
+		}
+		oldVersion, err := semver.ParseTolerant(old.Spec.Topology.Version)
+		if err != nil {
+			// NOTE: this should never happen. Nevertheless, handling this for extra caution.
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "topology", "version"),
+					new.Spec.Topology.Class,
+					"cannot be compared with the old version",
+				),
+			)
+		}
+		if inVersion.NE(semver.Version{}) && oldVersion.NE(semver.Version{}) && version.Compare(inVersion, oldVersion, version.WithBuildTags()) == -1 {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "topology", "version"),
+					new.Spec.Topology.Version,
+					"cannot be decreased",
+				),
+			)
+		}
+	}
+	// Check to see if the ClusterClass referenced in the Cluster currently exists.
+	if err := c.Client.Get(context.Background(), client.ObjectKey{Namespace: new.Namespace, Name: new.Spec.Topology.Class}, &clusterv1.ClusterClass{}); err != nil {
+		allErrs = append(
+			allErrs, field.Invalid(
+				field.NewPath("spec", "topology", "class"),
+				new.Name,
+				"clusterv1.ClusterClass could not be found"))
```

Comment:
```suggestion
				"ClusterClass could not be found"))
```
nit. clusterv1 is just our import alias, but I would prefer not adding any apiVersion here, as it's already clear what we mean without it and then we don't have to use a constant / or update the string on every apiVersion upgrade.

Reply:
Copy replace issue here